src/webui/abstractwebapplication.cpp

```diff
@@ -188,42 +203,13 @@ bool AbstractWebApplication::sessionInitialize()
     return false;
 }
 
-bool AbstractWebApplication::readFile(const QString& path, QByteArray &data, QString &type)
+QString AbstractWebApplication::getContentType(const QString &path)
 {
-    QString ext = "";
+    QString ext;
     int index = path.lastIndexOf('.') + 1;
     if (index > 0)
         ext = path.mid(index);
-
-    // find translated file in cache
-    if (translatedFiles_.contains(path)) {
-        data = translatedFiles_[path];
-    }
-    else {
-        QFile file(path);
-        if (!file.open(QIODevice::ReadOnly)) {
-            qDebug("File %s was not found!", qUtf8Printable(path));
-            return false;
-        }
-
-        data = file.readAll();
-        file.close();
-
-        // Translate the file
-        if ((ext == "html") || ((ext == "js") && !path.endsWith("excanvas-compressed.js"))) {
-            QString dataStr = QString::fromUtf8(data.constData());
-            translateDocument(dataStr);
-
-            if (path.endsWith("about.html") || path.endsWith("index.html") || path.endsWith("client.js"))
-                dataStr.replace("${VERSION}", QBT_VERSION);
-
-            data = dataStr.toUtf8();
-            translatedFiles_[path] = data; // cashing translated file
-        }
-    }
-
-    type = CONTENT_TYPE_BY_EXT[ext];
-    return true;
+    return CONTENT_TYPE_BY_EXT.value(ext, Http::CONTENT_TYPE_RAW);
```

Comment:
Why don't you use QMimeDatabase here?

Reply:
This is legacy code from Qt4 epoch.
Will use QMimeDatabase now.